File: code/smv-debugger/src/main/java/org/fbme/smvDebugger/integration/Fb2SmvService.java
package org.fbme.smvDebugger.integration;

import java.nio.file.Path;

import org.fbme.smvDebugger.commons.OS;
import org.fbme.smvDebugger.commons.OSUtils;

public class Fb2SmvService extends AbstractIntegrationService {
  private static final String SMV_FILE_EXTENSION = "smv";

  public Fb2SmvService(final Path fb2SmvBinaryPath) {
    super(fb2SmvBinaryPath);
  }

  public Path convertFbToSmv(final Path fbPath) {
    runProcess(fbPath);
    return newPathOf(fbPath, SMV_FILE_EXTENSION);
  }

  @Override
  protected String getCommand(final Path fbPath) {
    if (OSUtils.getOS() == OS.MAC_OS) {
      return MAC_MONO_FRAMEWORK + " " + binaryPath + " " + fbPath;
    }
    return binaryPath + " " + fbPath;
  }
}

File: code/smv-debugger/src/main/kotlin/org/fbme/smvDebugger/integration/Fb2SmvService.kt
package org.fbme.smvDebugger.integration

import org.fbme.smvDebugger.commons.OS
import org.fbme.smvDebugger.commons.OSUtils
import java.nio.file.Path

class Fb2SmvService(fb2SmvBinaryPath: Path?) : AbstractIntegrationService(fb2SmvBinaryPath) {
    fun convertFbToSmv(fbPath: Path?): Path {
        runProcess(fbPath!!)
        return newPathOf(fbPath, SMV_FILE_EXTENSION)
    }

    override fun getCommand(fbPath: Path): String {
        return if (OSUtils.oS == OS.MAC_OS) {
            MAC_MONO_FRAMEWORK + " " + binaryPath + " " + fbPath
        } else "$binaryPath $fbPath"
    }

    companion object {
        private const val SMV_FILE_EXTENSION = "smv"
    }
}
